### PR TITLE
[Integration][Azure] Support multi azure subscription and add generic cloudResource kind

### PR DESCRIPTION
**Pull request by Tankilevitch**
# Description

What -
- Added support for multi azure subscriptions
- Added generic cloudResource kind

Why -
- To reduce the need to install multiple azure installations
- To improve the onboarding of the integration and allow easier way to ingest more resources 

How -

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/azure/azure_integration/ocean.py [hunk 1]
```diff
@@ -47,29 +60,88 @@ async def resync_resource_groups(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
...
+    Re-syncs subscriptions, this is done separately because the subscriptions api is different from the other apis
+    """
     async with DefaultAzureCredential() as credential:
-        async with ResourceManagementClient(
-            credential=credential, subscription_id=get_integration_subscription_id()
-        ) as client:
-            async for resource_groups_batch in batch_resources_iterator(
-                client.resource_groups.list,
-                api_version=get_current_resource_config().selector.api_version,
+        async with SubscriptionClient(credential=credential) as client:
```
Comment (yairsimantov20) [inline integrations/azure/azure_integration/ocean.py:91]: combine with the one before

File: integrations/azure/azure_integration/ocean.py [hunk 2]
```diff
@@ -47,29 +60,88 @@ async def resync_resource_groups(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
...
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.CLOUD_RESOURCE)
+async def resync_cloud_resources(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    resource_kinds = typing.cast(
+        AzureCloudResourceSelector, get_current_resource_config().selector
+    ).resource_kinds
+    if not resource_kinds:
+        logger.warning(
+            "Resource kinds not found in port app config, skipping",
+        )
+        return
```
Comment (yairsimantov20) [inline integrations/azure/azure_integration/ocean.py:107]: make sure it wont happen with pydantic

File: integrations/azure/azure_integration/ocean.py [hunk 3]
```diff
@@ -47,29 +60,88 @@ async def resync_resource_groups(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
...
+    for resource_kind, resource_api_version in resource_kinds.items():
+        if is_sub_resource(resource_kind):
+            iterator_resync_method = resync_extension_resources
+        else:
+            iterator_resync_method = resync_base_resources
```
Comment (yairsimantov20) [inline integrations/azure/azure_integration/ocean.py:112]: function this

File: integrations/azure/azure_integration/ocean.py [hunk 4]
```diff
@@ -199,48 +219,58 @@ async def handle_events(cloud_event: CloudEvent) -> fastapi.Response:
...
+        if (
+            resource.kind == resource_type
+            and isinstance(resource.selector, AzureSpecificKindSelector)
+        )
+        or (
+            resource.kind == ResourceKindsWithSpecialHandling.CLOUD_RESOURCE
+            and isinstance(resource.selector, AzureCloudResourceSelector)
+            and resource_type in resource.selector.resource_kinds.keys()
+        )
     ]
```
Comment (yairsimantov20) [inline integrations/azure/azure_integration/ocean.py:236]: function this to should check

**Review by yairsimantov20 — APPROVED** [review]
🌊

File: integrations/azure/azure_integration/ocean.py (hunk 1 above)
Comment (Tankilevitch) [reply integrations/azure/azure_integration/ocean.py:91]: the credentials are used in the code afterwards not only in the subscription client, skipping this suggestion for now

**Pull request merged**